File: client/src/app/modules/miscellaneous/services/active.directory.group.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ActiveDirectoryGroup } from '../../miscellaneous/model/ActiveDirectoryGroup';
import { Employee } from '../../profile/model/Employee';

@Injectable({
  providedIn: 'root'
})
export class ActiveDirectoryGroupService extends AppRestService<ActiveDirectoryGroup> {

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getActiveDirectoryGroups() {
    return this.getList(new HttpParams(), "active-directory-groups");
  }

  addOrUpdateActiveDirectoryGroup(activeDirectoryGroup: ActiveDirectoryGroup) {
    return this.addOrUpdate(new HttpParams(), "active-directory-group", activeDirectoryGroup, "Enregistré", "Erreur lors de l'enregistrement");
  }

  isEmployeeInGroupList(employee: Employee, groups: ActiveDirectoryGroup[]) {
    if (groups && employee.adPath)
      for (let group of groups)
        if (employee.adPath.indexOf(group.activeDirectoryPath) >= 0)
          return true;
    return false;
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/ActiveDirectoryGroupService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.ActiveDirectoryGroup;

public interface ActiveDirectoryGroupService {
    public List<ActiveDirectoryGroup> getActiveDirectoryGroups();

    public ActiveDirectoryGroup getActiveDirectoryGroup(Integer id);

    public ActiveDirectoryGroup addOrUpdateActiveDirectoryGroup(ActiveDirectoryGroup activeDirectoryGroup);
}
